File: net472/FeatureDetection/IIS/MVCAppWithIISConfig/Middleware/AppShutDownHandler.cs
using System.Threading.Tasks;
using System.Web;


namespace TestMvcApplication
{
    public class AppShutDownHandler : IHttpHandler
    {

        public bool IsReusable
        {
            // Return false in case your Managed Handler cannot be reused for another request.  
            // Usually this would be false in case you have some state information preserved per request.  
            get { return true; }
        }


        public void ProcessRequest(HttpContext context)
        {
            string response = GenerateResponse(context);
            context.Response.Write($"<h1 style='Color:blue; font-size:15px;'>Our website is under maintainace. {response}</h1>");
        }

        public void TestMethod()
        {
            return;
        }
        private string GenerateResponse(HttpContext context)
        {
            return "This is a test handler";
        }
    }
}

File: net48/VBConsoleApp1/VBConsoleApp1/AppShutDownHandler.vb
Imports System.Web
Imports System.Web.Mvc

Public Class AppShutDownHandler
    Implements IHttpHandler

    Sub New(isReusable As Boolean)
        Me.IsReusable = isReusable
    End Sub

    Public Sub ProcessRequest(context As HttpContext) Implements IHttpHandler.ProcessRequest
        Dim response As String = GenerateResponse(context)
        context.Response.Write($"<h1 style='Color:blue; font-size:15px;'>Our website is under maintainace. {response}</h1>")
    End Sub

    Public ReadOnly Property IsReusable As Boolean Implements IHttpHandler.IsReusable

    Private Function GenerateResponse(context As HttpContext)
        Return "This is a test handler"
    End Function

End Class
